mlflow/tensorflow.py

```diff
@@ -298,54 +334,81 @@ def _load_pyfunc(path):
     """
     tf_saved_model_dir, tf_meta_graph_tags, tf_signature_def_key =\
         _get_and_parse_flavor_configuration(model_path=path)
-
-    tf_graph = tensorflow.Graph()
-    tf_sess = tensorflow.Session(graph=tf_graph)
-    with tf_graph.as_default():
-        signature_def = _load_tensorflow_saved_model(
-            tf_saved_model_dir=tf_saved_model_dir, tf_sess=tf_sess,
-            tf_meta_graph_tags=tf_meta_graph_tags, tf_signature_def_key=tf_signature_def_key)
-
-    return _TFWrapper(tf_sess=tf_sess, tf_graph=tf_graph, signature_def=signature_def)
+    if LooseVersion(tensorflow.__version__) < LooseVersion('2.0.0'):
+        tf_graph = tensorflow.Graph()
+        tf_sess = tensorflow.Session(graph=tf_graph)
+        with tf_graph.as_default():
+            signature_def = _load_tensorflow_saved_model(
+                tf_saved_model_dir=tf_saved_model_dir, tf_sess=tf_sess,
+                tf_meta_graph_tags=tf_meta_graph_tags, tf_signature_def_key=tf_signature_def_key)
+
+        return _TFWrapper(tf_sess=tf_sess, tf_graph=tf_graph, signature_def=signature_def)
+    else:
+        loaded_model = tensorflow.saved_model.load(export_dir=tf_saved_model_dir,
+                                                   tags=tf_meta_graph_tags)
+        return _TFWrapper(infer=loaded_model.signatures[tf_signature_def_key])
 
 
 class _TFWrapper(object):
     """
     Wrapper class that exposes a TensorFlow model for inference via a ``predict`` function such that
     ``predict(data: pandas.DataFrame) -> pandas.DataFrame``.
     """
-    def __init__(self, tf_sess, tf_graph, signature_def):
+    def __init__(self, infer=None, tf_sess=None, tf_graph=None, signature_def=None):
         """
+        For TensorFlow versions >= 2.0.0:
+        :param infer: Tensorflow function returned by a saved model that is used for inference.
+
+        For TensorFlow versions < 2.0.0:
         :param tf_sess: The TensorFlow session used to evaluate the model.
         :param tf_graph: The TensorFlow graph containing the model.
         :param signature_def: The TensorFlow signature definition used to transform input dataframes
                               into tensors and output vectors into dataframes.
         """
-        self.tf_sess = tf_sess
-        self.tf_graph = tf_graph
-        # We assume that input keys in the signature definition correspond to input DataFrame column
-        # names
-        self.input_tensor_mapping = {
+        self.infer = infer
+        if LooseVersion(tensorflow.__version__) < LooseVersion('2.0.0'):
```

Comment:
Instead of having an `if/else` version check in each method here, maybe it'd be easier to just have separate `_TF1Wrapper` and `_TF2Wrapper` classes & do the version check / return the appropriate class from `_load_pyfunc`?